lib/create-archive.js

```diff
@@ -0,0 +1,26 @@
+const raf = require('random-access-file')
+const encoding = require('dat-encoding')
+const assert = require('assert')
+
+module.exports = createArchive
+
+// create a new archive
+// (obj, obj{str, str, fn}) -> obj
+function createArchive (drive, { path, key, isFile }) {
+  assert.ok(drive, 'lib/create-archive: drive is not defined')
+  assert.ok(path, 'lib/create-archive: path is not defined')
+  // TODO(yw): figure out why a key's not always passed - probs irrelevant
+  // once we start using dat-js
+  // assert.ok(key, 'lib/create-archive: key is not defined')
+  assert.ok(isFile, 'lib/create-archive: isFile is not defined')
```

Comment:
if isFile is false, this always throws. So if you share a directory, it will crash. isFile is a boolean. It probably should check `isFile !== undefined` or something like that.